examples/cbr-multi-service-profile/main.tf

```diff
@@ -0,0 +1,183 @@
+##############################################################################
+# Get Cloud Account ID
+##############################################################################
+
+data "ibm_iam_account_settings" "iam_account_settings" {
+}
+
+##############################################################################
+# Resource Group
+##############################################################################
+
+module "resource_group" {
+  source = "git::https://github.com/terraform-ibm-modules/terraform-ibm-resource-group.git?ref=v1.0.5"
+  # if an existing resource group is not set (null) create a new one using prefix
+  resource_group_name          = var.resource_group == null ? "${var.prefix}-resource-group" : null
+  existing_resource_group_name = var.resource_group
+}
+
+##############################################################################
+# VPC
+##############################################################################
+resource "ibm_is_vpc" "example_vpc" {
+  name           = "${var.prefix}-vpc"
+  resource_group = module.resource_group.resource_group_id
+  tags           = var.resource_tags
+}
+
+resource "ibm_is_public_gateway" "testacc_gateway" {
+  name           = "${var.prefix}-pgateway"
+  vpc            = ibm_is_vpc.example_vpc.id
+  zone           = "${var.region}-1"
+  resource_group = module.resource_group.resource_group_id
+}
+
+resource "ibm_is_subnet" "testacc_subnet" {
+  name                     = "${var.prefix}-subnet"
+  vpc                      = ibm_is_vpc.example_vpc.id
+  zone                     = "${var.region}-1"
+  public_gateway           = ibm_is_public_gateway.testacc_gateway.id
+  total_ipv4_address_count = 256
+  resource_group           = module.resource_group.resource_group_id
+}
+
+##############################################################################
+# CBR zone & rule creation
+##############################################################################
+
+locals {
+  zone_list = [{
+    name             = "${var.prefix}-cbr-zone1"
+    account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+    zone_description = "cbr-zone1-terraform"
+    addresses = [{
+      type  = "vpc", # to bind a specific vpc to the zone
+      value = resource.ibm_is_vpc.example_vpc.crn,
+    }]
+    },
+    {
+      name             = "${var.prefix}-cbr-zone2"
+      account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+      zone_description = "cbr-zone2-terraform"
+      addresses = [{
+        type = "serviceRef" # to bind a service reference type should be 'serviceRef'
+        ref = {
+          account_id   = data.ibm_iam_account_settings.iam_account_settings.account_id
+          service_name = "directlink" # secrets manager service reference.
```

Comment:
No it would not fail.  Zones can be created upfront as per the architecture even if the service does not exist.  With that consideration, above infrastructure does not provision a DirectLink service.

Reply:
This was discussed and decided factors are considered for the new commit made.  So resolving this comment.